src/theory/builtin/proof_checker.cpp

```diff
@@ -187,48 +195,60 @@ bool BuiltinProofRuleChecker::getSubstitutionFor(Node exp,
   return ret;
 }
 
-Node BuiltinProofRuleChecker::applySubstitution(Node n, Node exp, MethodId ids)
+Node BuiltinProofRuleChecker::applySubstitution(Node n,
+                                                Node exp,
+                                                MethodId ids,
+                                                MethodId ida)
 {
-  std::vector<TNode> vars;
-  std::vector<TNode> subs;
-  std::vector<TNode> from;
-  if (!getSubstitutionFor(exp, vars, subs, from, ids))
-  {
-    return Node::null();
-  }
-  Node ns = n;
-  // apply substitution one at a time, in reverse order
-  for (size_t i = 0, nvars = vars.size(); i < nvars; i++)
-  {
-    TNode v = vars[nvars - 1 - i];
-    TNode s = subs[nvars - 1 - i];
-    Trace("builtin-pfcheck-debug")
-        << "applySubstitution (" << ids << "): " << v << " -> " << s
-        << " (from " << exp << ")" << std::endl;
-    ns = ns.substitute(v, s);
-  }
-  return ns;
+  std::vector<Node> expv{exp};
+  return applySubstitution(n, expv, ids, ida);
 }
 
 Node BuiltinProofRuleChecker::applySubstitution(Node n,
                                                 const std::vector<Node>& exp,
-                                                MethodId ids)
+                                                MethodId ids,
+                                                MethodId ida)
 {
-  Node curr = n;
-  // apply substitution one at a time, in reverse order
+  // Build a simultaneous subsitution, built in reverse order. This is done
+  // for performance reasons, so that only one Node::substitute call is made.
```

Comment:
the sequential case applies Node::substitute vars.size() times. Clarify.

Reply:
This comment is outdated, ill remove.